### PR TITLE
mapKeys for Map and convert for Format[T]

### PR DESCRIPTION
**Pull request by martinraison**
@stkem @leogrim @andrewconner Suggesting two super-simple convenience methods that I've always been missing: `mapKeys` for Map (same as `mapValues` but for keys) and `convert` for Format (to transform a Format[A] into a Format[B])

@lawzlo I think this should allow you to simplify your MapFormatUtil object quite a bit:

```
import com.keepit.common.collection._
import com.keepit.common.json._

object MapFormatUtil {
  implicit def scoreTypeMapFormat(implicit f: Format[Map[String, Float]]): Format[Map[ScoreType, Float]] =
    f.convert(_.mapKeys(ScoreType.withName _), _.mapKeys(_.toString))
}
```

### CODE REVIEW

**Comment by lawzlo** [comment]
That's awesome!

File: kifi-backend/modules/common/app/com/keepit/common/json/package.scala [hunk 1]
```diff
@@ -21,4 +21,14 @@ object JsonFormatters {
   implicit def tuple2Format[A, B](implicit reads: Reads[(A, B)], writes: Writes[(A, B)]) = {
     Format(reads, writes)
   }
+
+  implicit class PimpedFormat[A](format: Format[A]) {
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/json/package.scala:25]: sweet!

File: kifi-backend/modules/common/app/com/keepit/common/collection/package.scala [hunk 2]
```diff
@@ -2,13 +2,15 @@ package com.keepit.common
 
 package object collection {
 
-  implicit class PimpedMap[A, B](x: Map[A, B]) {
+  implicit class PimpedMap[A, B](val m: Map[A, B]) {
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/common/collection/package.scala]: why the `val`?

Comment (martinraison) [inline kifi-backend/modules/common/app/com/keepit/common/collection/package.scala]: because I forgot it here :)

**Pull request merged**